src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -0,0 +1,160 @@
+        private readonly CosmosReviewRepository _reviewsRepository;
+        private readonly BlobCodeFileRepository _codeFileRepository;
+        private readonly DevopsArtifactRepository _devopsArtifactRepository;
+
+        public PullRequestManager(
+            ReviewManager reviewManager,
+            CosmosReviewRepository reviewsRepository,
+            CosmosPullRequestsRepository pullRequestsRepository,
+            BlobCodeFileRepository codeFileRepository,
+            DevopsArtifactRepository devopsArtifactRepository,
+            IConfiguration configuration
+            )
+        {
+            _reviewManager = reviewManager;
+            _pullRequestsRepository = pullRequestsRepository;
+            _configuration = configuration;
+            _reviewsRepository = reviewsRepository;
+            _codeFileRepository = codeFileRepository;
+            _devopsArtifactRepository = devopsArtifactRepository;
+            _githubClient.Credentials = new Credentials(_configuration["github-access-token"]);
+        }
+
+        // API change detection for PR will pull artifact from devops artifact
+        public async Task DetectApiChanges(string buildId, string artifactName, string filePath, int prNumber, string commitSha, string repoName)
+        {
+            var requestTelemetry = new RequestTelemetry { Name = "Detecting API changes for PR: " + prNumber };
+            var operation = _telemetryClient.StartOperation(requestTelemetry);
+            try
+            {
+                var pullRequestModel = await _pullRequestsRepository.GetPullRequestAsync(prNumber, repoName, filePath);
+                if (pullRequestModel == null)
+                {
+                    pullRequestModel = new PullRequestModel()
+                    {
+                        RepoName = repoName,
+                        PullRequestNumber = prNumber,
+                        FilePath = filePath
+                    };
+                }
+                else
+                {
+                    if (pullRequestModel.Commits.Any(c=> c== commitSha))
+                    {
+                        // PR commit is already processed. No need to reprocess it again.
+                        return;
+                    }
+                }
+                pullRequestModel.Commits.Add(commitSha);
+
+                using var stream = await _devopsArtifactRepository.DownloadPackageArtifact(repoName, buildId, artifactName, filePath);
+                if (stream != null)
+                {
+                    using var memoryStream = new MemoryStream();
+                    var codeFile = await _reviewManager.CreateCodeFile(Path.GetFileName(filePath), stream, false, memoryStream);
+                    var apiDiff = await GetApiDiffFromAutomaticReview(codeFile);
+                    if (apiDiff != "")
+                    {
+                        var repoOwner = _configuration["default-source-repo-owner"];
```

Comment:
We might need to consider making this part of the repoName (i.e. Azure/azure-sdk-for-net). The reason I say that is there are other orgs for testing the specs changes and there are potentially forks "azure-sdk/azure-sdk-for-net" that we might end up needing to support.

Reply:
I can modify this to accept Github URL that can be parsed out to get owner or somehow use it as input to write comment.

Reply:
Actually I will pass it along with repoName as you suggested.

Reply:
Fixed